### PR TITLE
Uniquely index transactions

### PR DESCRIPTION
**Pull request by anler**
## What

* Add unit tests for `index_transactions` method
* Uniquely index transactions, that is, if a block is indexed twice its transactions aren't indexed/duplicated anymore in the database
* Add a new method `contains` to `Database` interface

### CODE REVIEW

File: wallet/src/repository/wallet/mod.rs [hunk 1]
```diff
@@ -514,6 +517,7 @@ where
             batch.put(keys::account_balance(account), new_balance)?;
             batch.put(keys::account_utxo_set(account), db_utxo_set)?;
             batch.put(keys::transaction_next_id(account), txn_next_id)?;
+            batch.put(keys::transactions_index(txn_hash), ())?;
```
Comment (tmpolaczyk) [inline wallet/src/repository/wallet/mod.rs]: Could you add a comment here?

```
// Mark transaction as indexed so we don't accidentally index it twice
```

Comment (anler) [reply wallet/src/repository/wallet/mod.rs]: Now I'm associating the transaction id to its hash, so I believe the comment is no longer needed

File: wallet/src/repository/wallet/tests/factories.rs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        )
+        .unwrap();
+
+    let wallet = Wallet::unlock(db.clone(), params, engine).unwrap();
+
+    (wallet, db)
+}
+
+pub fn pkh() -> types::PublicKeyHash {
+    let bytes: [u8; 20] = rand::random();
```
Comment (tmpolaczyk) [inline wallet/src/repository/wallet/tests/factories.rs]: Non-deterministic tests? :eyes:

Comment (anler) [reply wallet/src/repository/wallet/tests/factories.rs]: hahaha not really, these are pretty common for generating random data for tests, think of
[faker](https://github.com/paraseba/faker) or its Rust equivalent [faker-rs](https://github.com/cksac/fake-rs)

**Pull request merged**